### ISSUE TITLE
Fatal Exception: java.lang.ClassCastException

### ISSUE THREAD

**Issue by xyzwilliamxyz** [AzureAD/microsoft-authentication-library-for-android#1243]
**Describe the bug**
We're having this exception reported by crashlytics on the latest MSAL version (2.0.4):

```
Fatal Exception: java.lang.ClassCastException: java.util.concurrent.ExecutionException cannot be cast to com.microsoft.identity.common.exception.BaseException
       at com.microsoft.identity.client.PublicClientApplication$16.onError(PublicClientApplication.java:1738)
       at com.microsoft.identity.common.internal.controllers.CommandDispatcher.commandCallBackOnError(CommandDispatcher.java:256)
       at com.microsoft.identity.common.internal.controllers.CommandDispatcher.access$900(CommandDispatcher.java:74)
       at com.microsoft.identity.common.internal.controllers.CommandDispatcher$2$1.run(CommandDispatcher.java:241)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:145)
       at android.app.ActivityThread.main(ActivityThread.java:5835)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1399)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1194)
```
it seems to be a cast exception caused by some newly thrown exception. Could you guys have a look?

**Smartphone (please complete the following information):**
 - Device: Galaxy S5
 - Android Version: 5.0
 - MSAL Version: 2.0.4

**Comment by iambmelt** [comment]
Thanks @xyzwilliamxyz -- we'll take a look and ship a fix in an upcoming release

**Comment by gitaumoses4** [comment]
HI @xyzwilliamxyz  it's not clear to me why the exception is being thrown, as the issue was fixed before the release of MSAL Version 2.0.4

Here's the pull request: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1121

Are you able to share more details on reproducing the error?

**Comment by xyzwilliamxyz** [comment]
hi @gitaumoses4 , it's indeed on 2.0.4. All we have is the stacktrace.

**Comment by iambmelt** [comment]
Thanks for confirming @xyzwilliamxyz 

@gitaumoses4 and I took a peek at this together; we have indeed developed a fix for this crash already in https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1121 and it will be released in a _future_ release of MSAL. This fix was developed for `common@3.0.7`; MSAL `2.0.4` consumes `3.0.6` -- that is why you continue to see this crash in your telemetry.

This fix will ship in the January 2021 timeframe

**Comment by leonardochiaranda** [comment]
@iambmelt Thanks for the release date info. We'll keep track of it when the time comes.

**Comment by iambmelt** [comment]
@xyzwilliamxyz 

This fix is available in our most recent release of MSAL 2.0.5. Please feel free to reopen or file a new issues if you continue to experience problems after updating to the latest version
